zebrad/src/components/mempool.rs

```diff
@@ -365,11 +365,39 @@ impl Service<Request> for Mempool {
             if let Some(TipAction::Grow { block }) = tip_action {
                 tracing::trace!(block_height = ?block.height, "handling blocks added to tip");
 
+                let spent_outpoints = block
+                    .transactions
+                    .iter()
+                    .flat_map(|tx| tx.spent_outpoints())
+                    .collect();
+
+                let sprout_nullifiers = block
+                    .transactions
+                    .iter()
+                    .flat_map(|transaction| transaction.sprout_nullifiers())
+                    .collect();
+                let sapling_nullifiers = block
+                    .transactions
+                    .iter()
+                    .flat_map(|transaction| transaction.sapling_nullifiers())
+                    .collect();
+                let orchard_nullifiers = block
+                    .transactions
+                    .iter()
+                    .flat_map(|transaction| transaction.orchard_nullifiers())
+                    .collect();
+
                 // Cancel downloads/verifications/storage of transactions
                 // with the same mined IDs as recently mined transactions.
                 let mined_ids = block.transaction_hashes.iter().cloned().collect();
                 tx_downloads.cancel(&mined_ids);
                 storage.remove_same_effects(&mined_ids);
```

Comment:
Let's cache the rejections of the mined transaction IDs using a new `Mined` error?

This is an existing bug, but fixing it will make the mempool faster, because the transactions won't be re-downloaded and re-validated.

Reply:
Do we have an issue number for this existing bug? If so, let's link it in to this PR and/or split the fix into a new PR.

Reply:
> Let's cache the rejections of the mined transaction IDs using a new `Mined` error? ("transaction was committed to the best chain")

I thought it might as well reject every mined id from blocks on the chain, let me know if it should only reject those that were removed from the verified transactions.

> Do we have an issue number for this existing bug? If so, let's link it in to this PR and/or split the fix into a new PR.

It's a small change and not worth creating an issue for in my opinion.

Reply:
> > Let's cache the rejections of the mined transaction IDs using a new `Mined` error? ("transaction was committed to the best chain")
> 
> I thought it might as well reject every mined id from blocks on the chain, let me know if it should only reject those that were removed from the verified transactions.

Yes, this is better, just in case our mempool didn't have those transactions in it yet.